### PR TITLE
[MIG][13.0] l10n_fi_invoice_numbers: Migration to 13.0

### PR DESCRIPTION
**Pull request by Menestrels**
Relates to the task: https://odoo.web-veistamo.fi/web#id=12764&view_type=form&model=project.task&menu_id=192&action=201

### CODE REVIEW

File: l10n_fi_invoice_numbers/README.md [hunk 1]
```diff
@@ -0,0 +1,6 @@
+Finnish Invoice
+===============
+
+#### Version: 1.2.0
```
Comment (aisopuro) [inline l10n_fi_invoice_numbers/README.md:4]: We can probably remove this, we already have a version in the manifest.

File: l10n_fi_invoice_numbers/models/account_move.py [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+    _inherit = 'account.move'
+
+    @api.depends('name', 'state')
+    def _compute_ref_number(self):
+        for invoice in self:
+            if invoice.name:
+                invoice_number = re.sub(r'\D', '', invoice.name)
+                checksum = sum((7, 3, 1)[idx % 3] * int(val)
+                               for idx, val in enumerate(invoice_number[::-1]))
+                invoice.ref_number = invoice_number + str((10 - (checksum % 10)) % 10)
```
Comment (aisopuro) [inline l10n_fi_invoice_numbers/models/account_move.py:40]: This calculation already seems to be done in the `l10n_fi_payment_reference` module: https://github.com/OCA/l10n-finland/blob/11.0/l10n_fi_payment_reference/models/account_invoice.py#L53

This module should either:
- Use that module so the logic isn't duplicated unnecessarily
- Have a comment or other explanation of why it needs to re-calculate the finnish reference

As part of this you should consider whether it is necessary to skip RF-type references for some reason. This module does not support them, but the `l10n_fi_payment_reference` module does. Is that significant?

File: l10n_fi_invoice_numbers/__manifest__.py [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+#    published by the Free Software Foundation, either version 3 of the
+#    License, or (at your option) any later version.
+#
+#    This program is distributed in the hope that it will be useful,
+#    but WITHOUT ANY WARRANTY; without even the implied warranty of
+#    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+#    GNU Affero General Public License for more details.
+#
+#    You should have received a copy of the GNU Affero General Public License
+#    along with this program.  If not, see <http://www.gnu.org/licenses/>.
```
Comment (aisopuro) [inline l10n_fi_invoice_numbers/__manifest__.py:17]: This seems to be license text, isn't it already covered by the `LICENSE` file?

File: l10n_fi_invoice_numbers/models/account_move.py [hunk 4]
```diff
@@ -0,0 +1,63 @@
...
+                invoice.ref_number = False
+
+    invoice_number = fields.Char(
+        'Invoice number',
+        compute='_compute_ref_number',
+        store=True,
+        help=_('Identifier number used to refer to this invoice in '
+               'accordance with https://www.fkl.fi/teemasivut/sepa/'
+               'tekninen_dokumentaatio/Dokumentit/kotimaisen_viitte'
+               'en_rakenneohje.pdf')
```
Comment (aisopuro) [inline l10n_fi_invoice_numbers/models/account_move.py:53]: This link no longer works, we should update it to this:
```
https://www.finanssiala.fi/maksujenvalitys/dokumentit/kotimaisen_viitteen_rakenneohje.pdf
```

File: l10n_fi_invoice_numbers/models/account_move.py (hunk 2 above)
Comment (aisopuro) [reply l10n_fi_invoice_numbers/models/account_move.py:40]: It is possible this module is no longer required: the invoice template could just directly use the `payment_reference` field and separate the checksum-digit from it as necessary.

**Review by aisopuro — CHANGES_REQUESTED** [review]
My main concern is that this module seems to overlap a lot with `l10n_fi_payment_reference`: https://github.com/OCA/l10n-finland/blob/11.0/l10n_fi_payment_reference

File: l10n_fi_invoice_numbers/models/account_move.py (hunk 2 above)
Comment (mlaitinen) [reply l10n_fi_invoice_numbers/models/account_move.py:40]: And not even the `payment_reference` field, but the `invoice_payment_ref` that Odoo has introduced https://github.com/odoo/odoo/blob/31890c4e676f7f4f65221ca66bfd66cc0ceedecc/addons/account/models/account_move.py#L200

**Comment by mlaitinen** [comment]
Closing this PR as we don't need this module in 13

**Pull request closed without merging**